test/controllers/deploys_controller_test.rb

```diff
@@ -216,6 +215,90 @@
     unauthorized :delete, :destroy, project_id: :foo, id: 1
   end
 
+  as_a_viewer do
+    before do
+      Deploy.delete_all
+      Job.delete_all
+      cmd = 'cap staging deploy'
+      project = Project.first
+      job_def =  {project_id: project.id, command: cmd, status: nil, user_id: admin.id}
+      status = [
+        {status: 'failed', production: true },
+        {status: 'running', production: true},
+        {status:'succeeded', production: true},
+        {status:'succeeded', production: false}
+      ]
+
+      status.each do |stat|
+        job_def[:status] = stat[:status]
+        job = Job.create!(job_def)
+        Deploy.create!( {
+          stage_id: Stage.find_by_production(stat[:production]).id,
+          reference: 'reference',
+          job_id: job.id
+        } )
+      end
+    end
+
+    after do
+      Deploy.delete_all
+    end
+
+    describe "finds all deploys for a deployer" do
+      it "returns a 200" do
+        get :search, format: "json"
+        assert_response :ok
+      end
+
+      it "renders csv" do
+        get :search, format: "csv"
+        assert_equal "text/csv", @response.content_type
+        assert_response :ok
+      end
+
+      it "returns no results" do
+        get :search, format: "json", deployer: 'jimmyjoebob'
+        assert_response :ok
+        @response.body.must_equal "{\"deploys\":\[\]}"
+      end
+
+      it "returns 4 deploys for user Admin" do
+        get :search, format: "json", deployer: 'Admin'
+        assert_response :ok
+        deploys = JSON.parse(@response.body)
+        deploys["deploys"].count.must_equal 4
+      end
+
+      it "returns 2 sucessful deploys" do
+        get :search, format: "json", status: 'succeeded'
+        assert_response :ok
+        deploys = JSON.parse(@response.body)
+        deploys["deploys"].count.must_equal 2
+      end
+
+      it "returns 1 running deploy" do
+        get :search, format: "json", status: 'running'
+        assert_response :ok
+        deploys = JSON.parse(@response.body)
+        deploys["deploys"].count.must_equal 1
+      end
+
+      it "returns 4 deploys for project: Project" do
+        get :search, format: "json", project_name: "Project"
+        assert_response :ok
+        deploys = JSON.parse(@response.body)
+        deploys["deploys"].count.must_equal 4
+      end
+
+      it "returns 1 non production deploys" do
+        get :search, format: "json", production: 0
```

Comment:
lots of logic and only 1 test ... suspicious ;)
test with false ... and maybe 1  too ...